lib/ssz_ex.ex

```diff
@@ -402,40 +406,48 @@ defmodule LambdaEthereumConsensus.SszEx do
     end
   end
 
-  defp decode_list(binary, basic_type, size, schema) do
+  defp decode_fixed_list(binary, basic_type, size) do
     fixed_size = get_fixed_size(basic_type)
 
-    if check_valid_list_size_prev_decode(schema, fixed_size, size, binary) do
-      with {:ok, decoded_list} = result <-
-             binary
-             |> decode_chunk(fixed_size, basic_type)
-             |> flatten_results(),
-           :ok <- check_valid_list_size(schema, size, length(decoded_list)) do
-        result
-      end
-    else
-      {:error, "invalid list binary size"}
+    with {:ok, decoded_list} = result <-
+           binary
+           |> decode_chunk(fixed_size, basic_type)
+           |> flatten_results(),
```

Comment:
Let's have these three lines inside a function called `def decode_fixed_collection(collection, size, base_type)`, this is the only actual reusable piece of code between the two functions. This new function should replace the `decode_chunk/3` helper.